src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -2196,6 +2196,21 @@ public FixableDiagnostic ProviderDeclarationViaImportKeywordIsDeprecated(Provide
                 warnInsteadOfError ? DiagnosticLevel.Warning : DiagnosticLevel.Error,
                 "BCP403",
                 $"The enclosing array expects elements of type \"{expectedType}\", but the array being spread contains elements of incompatible type \"{actualType}\".");
+
+            public ErrorDiagnostic ExtendsPathHasNotBeenSpecified() => new(
+                TextSpan,
+                "BCP404",
```

Comment:
As a rule of thumb, we should have at least test to cover each of these new diagnostic codes.

See https://github.com/Azure/bicep/blob/de5d94b594039e8d912047caf2f9c9b5b9c9e5dc/src/Bicep.Core.IntegrationTests/ParametersTests.cs#L296-L315 for an example of the necessary test structure.